### PR TITLE
Try to re-enable unit tests on travis-ci

### LINKED ISSUE #14723: stylo builds aren't working on TravisCI
```
$ ./mach test-stylo

   Compiling stylo_tests v0.0.1 (file:///home/travis/build/servo/servo/tests/unit/stylo)

warning: value assigned to `saw_before` is never read, #[warn(unused_assignments)] on by default

  --> /home/travis/build/servo/servo/tests/unit/stylo/sanity_checks.rs:50:9

   |

50 |     let mut saw_before = false;

   |         ^^^^^^^^^^^^^^

warning: value assigned to `saw_after` is never read, #[warn(unused_assignments)] on by default

  --> /home/travis/build/servo/servo/tests/unit/stylo/sanity_checks.rs:51:9

   |

51 |     let mut saw_after = false;

   |         ^^^^^^^^^^^^^

warning: struct is never used: `StylesheetLoader`, #[warn(dead_code)] on by default

 --> /home/travis/build/servo/servo/tests/unit/stylo/../../../ports/geckolib/stylesheet_loader.rs:9:1

  |

9 | pub struct StylesheetLoader;

  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: method is never used: `new`, #[warn(dead_code)] on by default

  --> /home/travis/build/servo/servo/tests/unit/stylo/../../../ports/geckolib/stylesheet_loader.rs:12:5

   |

12 |     pub fn new() -> Self {

   |     ^

    Finished debug [unoptimized + debuginfo] target(s) in 20.29 secs

     Running /home/travis/build/servo/servo/target/geckolib/debug/deps/stylo_tests-019e27a7cd146829

running 2 tests

test sanity_checks::assert_basic_pseudo_elements ... ok

test sanity_checks::assert_restyle_hints_match ... ok

test result: ok. 2 passed; 0 failed; 0 ignored; 0 measured

   Compiling style v0.0.1 (file:///home/travis/build/servo/servo/components/style)

LLVM ERROR: IO failure on output stream.

error: Could not compile `style`.

To learn more, run the command again with --verbose.

The command "./mach test-stylo" exited with 101.
```

### LINKED ISSUE #15076: Travis builds fail due to lack of space during test-unit

### PR DESCRIPTION
**Pull request by UK992**
After https://github.com/servo/servo/pull/16573 and https://github.com/servo/servo/pull/16593, issue https://github.com/servo/servo/issues/15076 _could_ be fixed.

Edit:
Closes #14723
Closes #15076

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/16767)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -13,16 +13,13 @@ matrix:
       script:
          - ./mach build -d --verbose
          - ./mach test-compiletest
-         # disabled due to #15076
-         #- ./mach test-unit
-         # disabled due to #14723
-         #- ./mach build-geckolib
-         #- ./mach test-stylo
+         - ./mach test-unit
+         - ./mach clean
```
Comment (jdm) [inline .travis.yml]: Why is the clean necessary?

Comment (jdm) [inline .travis.yml]: Oh, that frees up all the space used by the non-geckolib build, doesn't it?

Comment (UK992) [inline .travis.yml]: Yes.

**Comment by jdm** [comment]
This looks good to me if it's squashed.

**Comment by UK992** [comment]
Squashed.

Closes #14723
Closes #15076

**Pull request merged**